### PR TITLE
Don't conflate the status of a sliced field with its parent field

### LINKED ISSUE #2931: Different results from identical tracers when using NetCDFOutputWriter
The example below creates a nonhydrostatic model with 6 identical tracers, calculates the vertical flux of those tracers in an identical way for all 6, and then writes it all to NetCDF. In the same file, it writes a vertical (x-z) slice of the fluxes using the `indices` flag, and an y-average of them:

```julia
using Oceananigans
using Oceananigans.Units

grid = RectilinearGrid(size=(16, 16, 16), extent = (500, 500, 120))

n_tracers = 6
tracer_symbols = [ Symbol(:τ,i) for i in 1:n_tracers ]
model = NonhydrostaticModel(; grid, tracers = (tracer_symbols...,))
@info model

uᵢ(x, y, z) = 1e-2 * randn()
set!(model, w=uᵢ)

tracer_IC_odd(x, y, z) = sin(2π*z/grid.Lz)
for i in 1:n_tracers
    @info "Setting tracer $i"
    expression = Meta.parse("set!(model, τ$i=tracer_IC_odd)")
    eval(expression)
end

simulation = Simulation(model, Δt=30, stop_time=0.15hours)

u, v, w = model.velocities

wτ = NamedTuple(Symbol(:w, key) => Field(w*τ) for (key,τ) in pairs(model.tracers))

outputs_full = (; wτ...)

outputs_yavg = NamedTuple( Symbol(key, :_yavg)=>Average(val, dims=(2,)) for (key, val) in zip(keys(outputs_full), outputs_full) )

outputs_xz1 = merge(outputs_full, outputs_yavg)
simulation.output_writers[:xz1_writer] = NetCDFOutputWriter(model, outputs_xz1;
                       filename = "data/TEST.nc",
                       schedule = TimeInterval(simulation.stop_time),
                       verbose=true,
                       indices = (:, 1, :),
                       overwrite_existing = true,
                       )
run!(simulation)
```

While the outputs should be the same (since the tracers and their advection are identical), I get different results for the y-averaged fluxes for different tracers:

image

Note that, while similar, tracers α=1,3,4 are different from α=2,5,6. The difference isn't large in this example, but can be made larger with more complexity in the calculations.

A couple of notes:

- Changing the line `wτ = NamedTuple(Symbol(:w, key) => Field(w*τ) for (key,τ) in pairs(model.tracers))` to 
```julia
wτ = NamedTuple(Symbol(:w, key) => w*τ for (key,τ) in pairs(model.tracers))
```
gets rid of the issue. Although doing the above prevents a user from using `Field(op, data=scratch_data.data)` to save memory, which in some cases (my case for example) is very important.

- Not passing the `indices` flag (i.e. writing 3D fields instead slices) apparently gets rid of this issue.

Here, even though the obvious easy solution is to separate averages from slices when writing, a user wouldn't know that since this fails silently and the wrong results can be pretty subtle (as the example above hopefully illustrates). For example, it popped up in one of my simulations and it took me a while to even realize what was happening, let alone figure out the solution.

My main question is: is this expected behavior? If so, should we somehow warn users (or even throw an error) do avoid mistakes?

### PR DESCRIPTION
**Pull request by glwagner**
Closes #2931 

We could add a test...

### CODE REVIEW

**Review by tomchor — APPROVED** [review]
Tested and indeed this PR fixes the issue

**Comment by tomchor** [comment]
> Closes #2931
> 
> We could add a test...

A test would be good, but like you said it yourself the tests are already pretty resource-intensive so feel free to merge without a test if you think it wouldn't be worth it.

**Comment by navidcy** [comment]
@tomchor can you confirm that this PR doesn't show that issue?

**Comment by tomchor** [comment]
> @tomchor can you confirm that this PR doesn't show that issue?

Yes. I tested this with a couple different number of tracers and all was good

**Comment by navidcy** [comment]
OK, I'll try to resolve #2922 before we merge this.

@glwagner bump patch release?

**Comment by navidcy** [comment]
That commit was REALLY quick!

**Review by navidcy — APPROVED** [review]

**Comment by navidcy** [comment]
I suggest we wait for https://github.com/CliMA/Oceananigans.jl/issues/2922. I'm on it.

File: Project.toml
Comment (navidcy) [inline Project.toml:4]:
```suggestion
version = "0.79.4"
```

**Comment by navidcy** [comment]
Since we didn't release v0.79.4 yet, I suggest we wait for this to merge + some other bug fixes in PRs that are ready to be merged after tests pass and release after.

cc @tomchor, @glwagner, @simone-silvestri

**Comment by tomchor** [comment]
@glwagner @navidcy merge and register a new version?

**Pull request merged**
